slasher/rpc/detect_update_min_max_span.go

```diff
@@ -0,0 +1,103 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+type detect func(span uint64, spans *ethpb.MinMaxSpan, source uint64) uint64
+
+func detectMax(span uint64, spans *ethpb.MinMaxSpan, source uint64) uint64 {
+	maxSpan := uint64(spans.MaxSpan)
+	if maxSpan > span {
+		return maxSpan + source
+	}
+	return 0
+}
+
+func detectMin(span uint64, spans *ethpb.MinMaxSpan, source uint64) uint64 {
+	minSpan := uint64(spans.MinSpan)
+	if minSpan < span {
+		return minSpan + source
+	}
+	return 0
+}
+
+func (ss *Server) detectSpan(source, target, validatorIdx uint64, detectionFunc detect) (uint64, uint64, *ethpb.EpochSpanMap, error) {
+	span := target - source + 1
```

Comment:
To an inexperienced reader, it isn't clear exactly what a `span` is. I understand it as a span of epochs between target and source, but we should be explaining some basic information about what a target and source are I think at least in the godocs of the `DetectAndUpdateMaxSpan` or min span exported functions. Maybe we could rename this variable to `epochSpan`? or some other measure to denote it is a distance metric?

Reply:
Just mostly reviewing this from the standpoint of readability and maintainability.

Reply:
thanks, epochSpan sounds better

Reply:
will add a comment

Reply:
can you please take a look? i added comment and name changed